### PR TITLE
Return base64 + remove recId from ECDSA signatures

### PR DESCRIPTION
**Pull request by darioAnongba**
Return base64 + remove recId from ECDSA signatures

### CODE REVIEW

File: src/pkg/crypto/eddsa.go [hunk 1]
```diff
@@ -19,19 +18,3 @@ func NewBN254() (eddsa.PrivateKey, error) {
 	// Same hFunc should be used for sign and verify
 	return eddsa.GenerateKey(bytes.NewReader(seed))
 }
-
-func ImportBN256(importedPrivKey string) (eddsa.PrivateKey, error) {
```
Comment (ggarri) [inline src/pkg/crypto/eddsa.go:23]: why did you remove this method as part of this PR?

File: src/vault/use-cases/keys/create_key.go [hunk 2]
```diff
@@ -59,8 +59,8 @@ func (uc *createKeyUseCase) Execute(ctx context.Context, namespace, id, algo, cu
 			return nil, errors.InvalidParameterError(errMessage)
 		}
 
-		key.PrivateKey = hexutil.Encode(privKey.Bytes())
-		key.PublicKey = hexutil.Encode(privKey.Public().Bytes())
+		key.PrivateKey = base64.StdEncoding.EncodeToString(privKey.Bytes())
```
Comment (ggarri) [inline src/vault/use-cases/keys/create_key.go:62]: I think it would be more suitable, as we use http communication, to use `base64.urlEncoding` as Azure did for its interfaces:
```
// JSONWebKey as of http://tools.ietf.org/html/draft-ietf-jose-json-web-key-18
type JSONWebKey struct {
	// Kid - Key identifier.
	Kid *string `json:"kid,omitempty"`
	// Kty - JsonWebKey Key Type (kty), as defined in https://tools.ietf.org/html/draft-ietf-jose-json-web-algorithms-40. Possible values include: 'EC', 'ECHSM', 'RSA', 'RSAHSM', 'Oct'
	Kty    JSONWebKeyType `json:"kty,omitempty"`
	KeyOps *[]string      `json:"key_ops,omitempty"`
	// N - RSA modulus. (a URL-encoded base64 string)
	N *string `json:"n,omitempty"`
	// E - RSA public exponent. (a URL-encoded base64 string)
	E *string `json:"e,omitempty"`
	// D - RSA private exponent, or the D component of an EC private key. (a URL-encoded base64 string)
	D *string `json:"d,omitempty"`
	// DP - RSA private key parameter. (a URL-encoded base64 string)
	DP *string `json:"dp,omitempty"`
	// DQ - RSA private key parameter. (a URL-encoded base64 string)
	DQ *string `json:"dq,omitempty"`
	// QI - RSA private key parameter. (a URL-encoded base64 string)
	QI *string `json:"qi,omitempty"`
	// P - RSA secret prime. (a URL-encoded base64 string)
	P *string `json:"p,omitempty"`
	// Q - RSA secret prime, with p < q. (a URL-encoded base64 string)
	Q *string `json:"q,omitempty"`
	// K - Symmetric key. (a URL-encoded base64 string)
	K *string `json:"k,omitempty"`
	// T - HSM Token, used with 'Bring Your Own Key'. (a URL-encoded base64 string)
	T *string `json:"key_hsm,omitempty"`
	// Crv - Elliptic curve name. For valid values, see JsonWebKeyCurveName. Possible values include: 'P256', 'P384', 'P521', 'P256K'
	Crv JSONWebKeyCurveName `json:"crv,omitempty"`
	// X - X component of an EC public key. (a URL-encoded base64 string)
	X *string `json:"x,omitempty"`
	// Y - Y component of an EC public key. (a URL-encoded base64 string)
	Y *string `json:"y,omitempty"`
}
```

File: src/vault/use-cases/keys/create_key.go [hunk 3]
```diff
@@ -88,37 +88,48 @@ func (uc *createKeyUseCase) Execute(ctx context.Context, namespace, id, algo, cu
...
+	return key, nil
 }
 
 func (*createKeyUseCase) ecdsaSecp256k1(importedPrivKey string) (*ecdsa.PrivateKey, error) {
 	if importedPrivKey == "" {
-		key, err := crypto.NewSecp256k1()
+		key, err := crypto.GenerateKey()
 		if err != nil {
-			return key, errors.CryptoOperationError(err.Error())
+			return nil, err
```
Comment (ggarri) [inline src/vault/use-cases/keys/create_key.go:118]: isn't this a crypto operation error anymore?

File: src/vault/use-cases/keys/create_key.go [hunk 4]
```diff
@@ -69,8 +69,8 @@ func (uc *createKeyUseCase) Execute(ctx context.Context, namespace, id, algo, cu
 			return nil, errors.InvalidParameterError(errMessage)
 		}
 
-		key.PrivateKey = hex.EncodeToString(crypto2.FromECDSA(privKey))
-		key.PublicKey = hexutil.Encode(crypto2.FromECDSAPub(&privKey.PublicKey))
+		key.PrivateKey = base64.StdEncoding.EncodeToString(crypto.FromECDSA(privKey))
```
Comment (ggarri) [inline src/vault/use-cases/keys/create_key.go:72]: I think it would be good to encapsulate encode and decode operation under a util so we can easily move from one type to another without modifying it all over our codebase, WDYT?

File: src/pkg/crypto/eddsa.go (hunk 1 above)
Comment (darioAnongba) [reply src/pkg/crypto/eddsa.go:23]: It's only used in one place `create_key` and it's mixing decoding and crypto operation in the same function.

Comment (darioAnongba) [reply src/pkg/crypto/eddsa.go:23]: I removed the useless functions in `crypto/ecdsa` and `crypto/eddsa`

File: src/vault/use-cases/keys/create_key.go (hunk 2 above)
Comment (darioAnongba) [reply src/vault/use-cases/keys/create_key.go:62]: done

File: src/vault/use-cases/keys/create_key.go (hunk 3 above)
Comment (darioAnongba) [reply src/vault/use-cases/keys/create_key.go:118]: these errors are not used and they all translate to `InvalidParameterError` in the use case, we can simply return the error.

File: src/vault/use-cases/keys/create_key.go (hunk 4 above)
Comment (darioAnongba) [reply src/vault/use-cases/keys/create_key.go:72]: done

**Review by ggarri — APPROVED** [review]
